apps/researcher/src/app/[locale]/communities/[id]/page.tsx

```diff
@@ -0,0 +1,107 @@
+import {ChevronLeftIcon} from '@heroicons/react/24/solid';
+import {clerkClient, auth} from '@clerk/nextjs';
+import Link from 'next/link';
+import Image from 'next/image';
+import {getTranslations} from 'next-intl/server';
+import {JoinCommunityButton, EditCommunityButton} from './buttons';
+
+interface Props {
+  params: {id: string};
+}
+
+export default async function CommunityPage({params}: Props) {
+  const t = await getTranslations('Community');
+
+  let organization;
```

Comment:
Thinking out load: would it make sense to move this logic (lines 15 to 27) to a separate file? That way we (1) have less logic in this component, (2) make the logic testable, (3) hide the fact that we're using Clerk?